File: core/src/ar/edu/unq/games/arkanoid/actors/Ball.scala
package ar.edu.unq.games.arkanoid.actors

import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.math.Vector2
import com.badlogic.gdx.physics.box2d.BodyDef.BodyType
import com.badlogic.gdx.physics.box2d._

class Ball extends PhysicActor {
  val ball = new Texture("ball.png")
  val bodyType = BodyType.DynamicBody
  val initialPosition = new Vector2(200f, 300f)

  override def init(world: World): Unit = {
    super.init(world)
    createBody(world)
    val scalarVelocity: Float = -600
    body.setLinearVelocity(new Vector2(0, scalarVelocity))
  }

  //=================================================================

  def createBody(world: World): Unit = {
    initBody(world)

    val shape = new CircleShape()
    val radius: Float = ball.getWidth / 2
    shape.setRadius(radius)
    shape.setPosition(new Vector2(radius, radius))

    val fixtureDef = new FixtureDef()
    fixtureDef.shape = shape
    fixtureDef.density = 1f
    fixtureDef.friction = 0f
    fixtureDef.restitution = 1f
    body.setUserData(this)
    body.createFixture(fixtureDef)
    shape.dispose()
  }

  override def draw(spriteBatch: SpriteBatch): Unit = {
    spriteBatch.draw(ball, pos.x, pos.y)
  }

}

File: core/src/ar/edu/unq/games/arkanoid/scripts/Ball.java
package ar.edu.unq.games.arkanoid.scripts;

import com.badlogic.ashley.core.Entity;
import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.physics.box2d.World;
import com.uwsoft.editor.renderer.components.DimensionsComponent;
import com.uwsoft.editor.renderer.components.physics.PhysicsBodyComponent;
import com.uwsoft.editor.renderer.scripts.IScript;
import com.uwsoft.editor.renderer.utils.ComponentRetriever;

public class Ball implements IScript {

    private Entity ball;
    private DimensionsComponent dimensionsComponent;
    private PhysicsBodyComponent physicsBodyComponent;
    private boolean isMoving = false;

    private World world;

    private Vector2 speed;

    public Ball(World world) {
        this.world = world;
    }


    @Override
    public void init(Entity entity) {
        ball = entity;

//        dimensionsComponent = ComponentRetriever.get(entity, DimensionsComponent.class);
        physicsBodyComponent = ComponentRetriever.get(entity, PhysicsBodyComponent.class);

        speed = new Vector2(0, 10);

    }

    @Override
    public void act(float delta) {
        if (!isMoving) {
            isMoving = true;
            physicsBodyComponent.body.applyForce(speed, new Vector2(0, 0), true);
        }
    }

    @Override
    public void dispose() {

    }

}
